src/js/lib/monster.ui.js

```diff
@@ -976,60 +979,69 @@ define(function(require) {
 		// Set to 'monthly', it will always set the end date to be one month away from the start date (unless it's after "today")
 		// container: jQuery Object. Container of the datepickers
 		initRangeDatepicker: function(options) {
-			var self = this,
-				container = options.container,
+			var timezone;
+
+			if (_.has(monster, 'apps.auth.currentUser.timezone')) {
+				timezone = monster.apps.auth.currentUser.timezone;
+			} else if (_.has(monster, 'apps.auth.currentAccount.timezone')) {
+				timezone = monster.apps.auth.currentAccount.timezone;
+			} else {
+				timezone = moment.tz.guess();
+			}
+
+			var container = options.container,
 				range = options.range || 7,
 				initRange = options.initRange || options.range || 1,
 				inputStartDate = container.find('#startDate'),
 				inputEndDate = container.find('#endDate'),
-				now = new Date(),
-				today = new Date(now.getFullYear(), now.getMonth(), now.getDate(), 0, 0, 0, 0),
+				initDate = moment().tz(timezone),
+				today = moment().tz(timezone).set({hours: 0, minutes: 0, seconds: 0}),
 				startDate = _.get(
 					options,
 					'startDate',
-					new Date(now.getFullYear(), now.getMonth(), now.getDate(), 0, 0, 0, 0)
+					today
 					),
-				endDate = new Date(now.getFullYear(), now.getMonth(), now.getDate(), 23, 59, 59, 999);
+				endDate = moment().tz(timezone).set({hours: 23, minutes: 59, seconds: 59});
```

Comment:
Add milliseconds, just in case, to coincide with previous code.
```suggestion
				endDate = moment().tz(timezone).set({ hours: 23, minutes: 59, seconds: 59, milliseconds: 999 });
```

Reply:
```suggestion
				endDate = moment().tz(timezone).endOf('day');
```